planner/core/casetest/rule_derive_topn_from_window_test.go

```diff
@@ -82,3 +138,28 @@ func TestPushDerivedTopnPositive(t *testing.T) {
 		res.Check(testkit.Rows(output[i].Res...))
 	}
 }
+
+// Negative test when tidb_opt_derive_topn is off
+func TestPushDerivedTopnFlagOff(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("set tidb_opt_derive_topn=0")
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b int, primary key(b,a))")
```

Comment:
Can you add cases to test what will happen to pk with generated columns? Like `create table t (a int, b int as (a+1) stored, primary key(b,a));`

Reply:
What is the secanrio you intend to test? is it partition by a,b or is it partition by a,a1+1?

Reply:
Maybe both? Just to check it works well too with generated columns.

Reply:
@chrysan this is done along with previous comments in latest update.